src/controllers/resource-store/index.tsx

```diff
@@ -90,21 +118,29 @@ export const actions: Actions = {
       cached.accessedAt = getAccessedAt();
       dispatch(setResourceState(type, key, cached));
 
-      return cached;
+      return Promise.resolve(cached);
```

Comment:
Interestingly removing the somewhat unnecessary `async` on this function causes the `getKey()` to evaluate earlier than before.